poetry/masonry/builders/editable.py

```diff
@@ -197,21 +197,27 @@ def _add_dist_info(self, added_files: List[Path]) -> None:
         from poetry.core.masonry.builders.wheel import WheelBuilder
 
         added_files = added_files[:]
-
         builder = WheelBuilder(self._poetry)
 
-        dist_info_path = Path(builder.dist_info)
-        for dist_info in self._env.site_packages.find(
-            dist_info_path, writable_only=True
-        ):
-            if dist_info.exists():
-                self._debug(
-                    "  - Removing existing <c2>{}</c2> directory from <b>{}</b>".format(
-                        dist_info.name, dist_info.parent
+        # remove dist_info for all existing versions, in all candidate directories we can modify
+        dist_info_paths = [
+            Path(dist_info)
+            for site_package_dir in self._env.site_packages.writable_candidates
+            for dist_info in site_package_dir.glob(builder.dist_info_glob)
```

Comment:
Since the glob is only really required for an editable builder; you can construct it here.

Reply:
I can, sure. If I do, though, `return "{}-*.dist-info".format(escaped_name)` depends on implementation details (structure of format string, implementation of `escape_name(self._package.name)`), which I felt was iffy.